tiledb/dataframe_.py

```diff
@@ -82,129 +72,103 @@ def parse_tiledb_kwargs(kwargs):
     return parsed_args
 
 
-def is_nullable_extension_type(t):
-    import pandas as pd
-
-    return isinstance(
-        t,
-        (
-            pd.Int64Dtype,
-            pd.Int32Dtype,
-            pd.Int16Dtype,
-            pd.Int8Dtype,
-            pd.UInt64Dtype,
-            pd.UInt32Dtype,
-            pd.UInt16Dtype,
-            pd.UInt8Dtype,
-        ),
-    )
+@dataclass(init=False, frozen=True)
```

Comment:
Technically we currently support 3.6+, and only 3.6 and 3.7 work on TileDB-Cloud... Is the [dataclasses](https://pypi.org/project/dataclasses/) backport package sufficient here? If so that's fine to add as a dependency IMO.

Reply:
Hmm, looks like the 3.6 test was inadvertently dropped, which is why this didn't error out. We do still build wheels for it, so I'm going to add a test target back to the matrix.

Reply:
Yes I have used the dataclasses backport with 3.6 before, it's the same as the official (there used to be a bug but I see the fix has been backported too since last time I checked). Having said that I'm ok with dropping the dataclass if we don't want to add an optional dependency, it's merely a convenience.

Reply:
It can be conditional on 3.6 so I'm fine with adding the dep there.

Reply:
(There are other places I've wanted to use it)